enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverter.java

```diff
@@ -182,6 +184,15 @@ private UiExtensions convertUiExtension(
         }
     }
 
+    private UiExtensions deserializeUiExtensions(final String uiJsonString, final OperationDetailResponse operationDetail) throws JsonProcessingException {
+        final UiExtensions uiExtensions = objectMapper.readValue(uiJsonString, UiExtensions.class);
+        if (uiExtensions.getPreApprovalScreen() != null && uiExtensions.getPreApprovalScreen().getType() == QR_SCAN && operationDetail.getTotp() == null) {
+            logger.debug("Template for operation ID: {} is configured to use pre-approval screen QR_SCAN, but OTP was not created", operationDetail.getId());
```

Comment:
I suggest to use a higher level than `debug` for this log message.

Reply:
Increased log level to `info`. It is a legal use-case, creation of OTP may be suppressed, so I avoid increasing to `warn`.